dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/dispatch/ExecutorDispatcher.java

```diff
@@ -86,13 +86,13 @@ public Boolean dispatch(final ExecutionContext context) throws ExecuteException
         if (StringUtils.isEmpty(host.getAddress())) {
             logger.warn("fail to execute : {} due to no suitable worker, current task needs worker group {} to execute",
                     context.getCommand(), context.getWorkerGroup());
-            return false;
+            throw new ExecuteException("no suitable worker");
```

Comment:
```suggestion
            throw new WorkerGroupNotFoundException("no suitable worker");
```

Reply:
We cannot throw `WorkerGroupNotFoundException`  here, since the worker group may exist, just no suitable worker(Worker is overload).